models/models.js

```diff
@@ -22,6 +22,28 @@ exports.selectEndpoints = () => {
 
 }
 
+exports.selectArticles = (req, res, next) => {
+    return db.query(`SELECT 
+                        articles.author,
+                        articles.title,
+                        articles.article_id,
+                        articles.topic,
+                        articles.created_at,
+                        articles.votes,
+                        articles.article_img_url,
+                        COALESCE(comment_counts.comment_count, 0) AS comment_count
+                    FROM articles
+                    LEFT JOIN (
+                        SELECT article_id, COUNT(article_id) AS comment_count 
+                        FROM comments 
+                        GROUP BY article_id
+                    ) AS comment_counts
+                    ON articles.article_id = comment_counts.article_id
+                    ORDER BY articles.created_at DESC`).then(({rows}) => {
+        return rows
+    })
+}
```

Comment:
It's a bit too complicated. Not sure if it works as expected, but there's definitely an easier way to do it. Please check the notes https://l2c.northcoders.com/courses/sd-notes/back-end#sectionId=aggregate-functions,step=intro as your reference